plugins/parsers/dropwizard/parser.go

```diff
@@ -0,0 +1,239 @@
+	metrics = p.readDWMetrics("timer", dwr["timers"], metrics, metricTime)
+
+	jsonTags := p.readTags(buf)
+
+	// fill json tags first
+	if len(jsonTags) > 0 {
+		for _, m := range metrics {
+			for k, v := range jsonTags {
+				// only set the tag if it doesn't already exist:
+				if !m.HasTag(k) {
+					m.AddTag(k, v)
+				}
+			}
+		}
+	}
+	// fill default tags last
+	if len(p.DefaultTags) > 0 {
+		for _, m := range metrics {
+			for k, v := range p.DefaultTags {
+				// only set the default tag if it doesn't already exist:
+				if !m.HasTag(k) {
+					m.AddTag(k, v)
+				}
+			}
+		}
+	}
+
+	return metrics, nil
+}
+
+// InitTemplating initializes the templating support
+func (p *Parser) InitTemplating() error {
+	if len(p.Templates) > 0 {
+		defaultTemplate, _ := templating.NewDefaultTemplateWithPattern("measurement*")
+		templateEngine, err := templating.NewEngine(p.Separator, defaultTemplate, p.Templates)
+		p.templateEngine = templateEngine
+		return err
+	}
+	return nil
+}
+
+// ParseLine is not supported by the dropwizard format
+func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
+	return nil, fmt.Errorf("ParseLine not supported: %s, for data format: dropwizard", line)
+}
+
+// SetDefaultTags sets the default tags
+func (p *Parser) SetDefaultTags(tags map[string]string) {
+	p.DefaultTags = tags
+}
+
+func (p *Parser) readTags(buf []byte) map[string]string {
+
+	if p.TagsPath != "" {
+		var tagsBytes []byte
+		tagsResult := gjson.GetBytes(buf, p.TagsPath)
+		if tagsResult.Index > 0 {
+			tagsBytes = buf[tagsResult.Index : tagsResult.Index+len(tagsResult.Raw)]
+		} else {
+			tagsBytes = []byte(tagsResult.Raw)
+		}
+		var tags map[string]string
+		err := json.Unmarshal(tagsBytes, &tags)
+		if err != nil {
+			log.Printf("W! failed to parse tags from json path '%s': %s\n", p.TagsPath, err)
+		} else if len(tags) > 0 {
+			return tags
+		}
+	}
+
+	tags := make(map[string]string)
+	for tagKey, jsonPath := range p.TagPathsMap {
+		tags[tagKey] = gjson.GetBytes(buf, jsonPath).String()
+	}
+	return tags
+}
+
+func (p *Parser) parseTime(buf []byte) time.Time {
+
+	if p.TimePath != "" {
+		timeFormat := p.TimeFormat
+		if timeFormat == "" {
+			timeFormat = time.RFC3339
+		}
+		time, err := time.Parse(timeFormat, gjson.GetBytes(buf, p.TimePath).String())
+		if err == nil {
+			return time.UTC()
+		}
+	}
+	return time.Now().UTC()
+}
+
+func (p *Parser) unmarshalMetrics(buf []byte) (map[string]interface{}, error) {
+
+	var registryBytes []byte
+	if p.MetricRegistryPath != "" {
+		regResult := gjson.GetBytes(buf, p.MetricRegistryPath)
+		if regResult.Index > 0 {
+			registryBytes = buf[regResult.Index : regResult.Index+len(regResult.Raw)]
+		} else {
+			registryBytes = []byte(regResult.Raw)
+		}
+	} else {
+		registryBytes = buf
+	}
+	var jsonOut map[string]interface{}
+	err := json.Unmarshal(registryBytes, &jsonOut)
+	if err != nil {
+		err = fmt.Errorf("unable to parse dropwizard metric registry from JSON document, %s", err)
```

Comment:
I would like to see a more precise error message if the path is not found.

Reply:
done